brewtils/__init__.py

```diff
@@ -26,71 +35,70 @@ def get_easy_client(**kwargs):
-    :param kwargs: Keyword Arguments
-    :return:
+    Returns:
+        :dict: Parameters needed to make a connection to beergarden
     """
-    if kwargs.get(key, None) is not None:
-        return kwargs[key]
+    config = load_config(**kwargs)
+
+    return {key: config[key] for key in ('bg_host', 'bg_port', 'ssl_enabled', 'api_version',
+                                         'ca_cert', 'client_cert', 'url_prefix', 'ca_verify')}
 
-    for name in env_names:
-        if name in os.environ:
-            return os.environ[name]
 
-    return default
+def load_config(cli_args=None, **kwargs):
+    """Load configuration using Yapconf
 
+    Configuation will be loaded from these sources, with earlier sources having higher priority:
 
-def get_bg_connection_parameters(**kwargs):
-    """Parse the keyword arguments, search in the arguments, and environment for the values
+        1. ``**kwargs`` passed to this method
+        2. ``cli_args`` passed to this method
+        3. Environment variables using the ``BG_`` prefix
+        4. Default values in the brew_view specification
 
-    :param kwargs:
-    :return:
+    Args:
+        cli_args (list, optional): List of command line arguments for configuration loading
+        **kwargs: Additional configuration overrides
+
+    Returns:
+        :obj:`box.Box`: The resolved configuration object
     """
-    from brewtils.rest.client import RestClient
-    from brewtils.errors import BrewmasterValidationError
-
-    host = kwargs.pop('host', None) or os.environ.get('BG_WEB_HOST')
-    if not host:
-        raise BrewmasterValidationError('Unable to create a plugin without a beer-garden host. '
-                                        'Please specify one with bg_host=<host> or by setting the '
-                                        'BG_WEB_HOST environment variable.')
-
-    port = get_from_kwargs_or_env('port', ['BG_WEB_PORT'], '2337', **kwargs)
-
-    url_prefix = get_from_kwargs_or_env('url_prefix', ['BG_URL_PREFIX'], None, **kwargs)
-    url_prefix = normalize_url_prefix(url_prefix)
-
-    ssl_enabled = get_bool_from_kwargs_and_env('ssl_enabled', 'BG_SSL_ENABLED', **kwargs)
-    ca_verify = get_bool_from_kwargs_and_env('ca_verify', 'BG_CA_VERIFY', **kwargs)
-
-    api_version = kwargs.pop('api_version', RestClient.LATEST_VERSION)
-    ca_cert = get_from_kwargs_or_env('ca_cert', ['BG_CA_CERT', 'BG_SSL_CA_CERT'], None, **kwargs)
-    client_cert = get_from_kwargs_or_env('client_cert', ['BG_CLIENT_CERT', 'BG_SSL_CLIENT_CERT'],
-                                         None, **kwargs)
-
-    return {
-        'host': host,
-        'port': port,
-        'ssl_enabled': ssl_enabled,
-        'api_version': api_version,
-        'ca_cert': ca_cert,
-        'client_cert': client_cert,
-        'url_prefix': url_prefix,
-        'ca_verify': ca_verify
-    }
+    sources = []
+
+    if kwargs:
+        # Do a little kwarg massaging for backwards compatibility
```

Comment:
Do we want to be raising warnings here?

Reply:
I guess yeah, so we can remove this eventually.